src/rust/protocol/src/memcache/wire/request/parse.rs

```diff
@@ -0,0 +1,410 @@
+// Copyright 2021 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+use super::super::*;
+use crate::*;
+
+use core::slice::Windows;
+use std::convert::TryFrom;
+
+const MAX_BYTES: usize = usize::MAX / 2;
+const MAX_COMMAND_LEN: usize = 255;
```

Comment:
Given how `MAX_COMMAND_LEN` is used below, it is not inclusive of keys. So is this value too big? Something that doesn't have a `CRLF` _or_ space will be just the command verb, which is a mere 9 characters (`FLUSH_ALL`) tops? This puts the max command len at 10 characters if we factor in `\r` at the end (but no `\n` yet). So this constant should be something like 10-16 bytes?

This is in nitpicking territory, but having the command length slightly larger than key length gave me the initial impression that it includes the size of the key, which is not the case.

Reply:
Going with 16bytes to be generous here.